internal/clients/azure.go

```diff
@@ -65,6 +66,9 @@ var (
 	credentialsSourceUpbound                       xpv1.CredentialsSource = "Upbound"
 
 	upboundProviderIdentityTokenFile = "/var/run/secrets/upbound.io/provider/token"
+
+	// Round-robin counter for service principal selection
+	spCounter uint64
```

Comment:
nit: For the sake of clarity, what do you think about the following name:

```suggestion
	servicePrincipalCounter uint64
```

Reply:
same for here